### ISSUE TITLE
Re-evaluate labels and labelling conventions for data table v2

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#1579]
## Environment
Tested using NVDA 2017

## Issue
The labeling convention needs to be re-evalutated as it's caused some frustration during user testing.

The labels are overly long and are read on each sort of the table. 

_"If the table header is above a link -- I know what it's sorting"._ 😄 

![dec-20-2018 20-41-37](https://user-images.githubusercontent.com/40970507/50321693-b7b25000-0497-11e9-9558-899cf1bcc424.gif)

**Comment by joshblack** [comment]
What's the preferred label for this?

**Comment by dakahn** [comment]
closing -- will retest post launch
